### PR TITLE
764 read_input_data_county fails for 2023 start dates

### PR DESCRIPTION
**Pull request by HenrZu**
# Changes and Information

Please **briefly list the changes** made, additional Information and what the Reviewer should look out for:

- Adjusting the initialization in the vvs model since there were problems with newer starting dates. Which are caused since the intersection between vaccinated and recovered is not disjunct as assumed.
- A comparison for the h5 can be provided 

## Merge Request - Guideline Checklist

Please check our [git workflow](https://github.com/DLR-SC/memilio/wiki/git-workflow). Use the **draft** feature if the Pull Request is not yet ready to review.

### Checks by code author

- [x] Every addressed issue is linked (use the "Closes #ISSUE" keyword below)
- [x] New code adheres to [coding guidelines](https://github.com/DLR-SC/memilio/wiki/Coding-guidelines)
- [x] No large data files have been added (files should in sum not exceed 100 KB, avoid PDFs, Word docs, etc.)
- [x] Tests are added for new functionality and a local test run was successful
- [x] Appropriate **documentation** for new functionality has been added (Doxygen in the code and Markdown files if necessary)
- [x] Proper attention to licenses, especially no new third-party software with conflicting license has been added

### Checks by code reviewer(s)

- [x] Corresponding issue(s) is/are linked and addressed
- [x] Code is clean of development artifacts (no deactivated or commented code lines, no debugging printouts, etc.)
- [x] Appropriate **unit tests** have been added, CI passes and code coverage is acceptable (did not decrease)
- [x] No large data files added in the whole history of commits(files should in sum not exceed 100 KB, avoid PDFs, Word docs, etc.)

### CODE REVIEW

File: cpp/models/ode_secirvvs/parameters_io.h [hunk 1]
```diff
@@ -1097,9 +1105,27 @@ IOResult<void> export_input_data_county_timeseries(
...
+                                extrapolated_rki[county][day](
+                                    (size_t)InfectionState::InfectedSymptomsImprovedImmunityConfirmed +
+                                    age_group_offset) -
+                                extrapolated_rki[county][day]((size_t)InfectionState::InfectedSevereImprovedImmunity +
+                                                              age_group_offset) -
+                                extrapolated_rki[county][day]((size_t)InfectionState::InfectedCriticalImprovedImmunity +
+                                                              age_group_offset),
+                            std::max(0.0,
+                                     double(extrapolated_rki[county][day](
+                                         (size_t)InfectionState::SusceptibleImprovedImmunity + age_group_offset))));
```
Comment (mknaranja) [inline cpp/models/ode_secirvvs/parameters_io.h:1128]: I cannot comment below but is S_PI not lacking the Dead_PI change from above?

**Review by mknaranja — CHANGES_REQUESTED** [review]
I just had one comment. Then it could be merged.

File: cpp/models/ode_secirvvs/parameters_io.h (hunk 1 above)
Comment (HenrZu) [reply cpp/models/ode_secirvvs/parameters_io.h:1128]: In the context of extrapolation, only DeadNaive is currently used. All reported dead are written to this compartment.
 See
https://github.com/DLR-SC/memilio/blob/2de2900be8f493c57715f5636e9ad5f64ddcff43/cpp/models/ode_secirvvs/parameters_io.h#L1197
Above theres also an short explanation. 
For completeness, we could use the Dead_PI here as well, but that could lead to an additional source of errors (incorrectly initialized values there) and communicate a misunderstanding to the user that they think the variable is being used.

Comment (mknaranja) [reply cpp/models/ode_secirvvs/parameters_io.h:1128]: The problem is: You now introduced the Death_PI/II compartments to the read_input_data which should give exactly the same result as the extrapolation. This is now not consistent anymore.

Comment (HenrZu) [reply cpp/models/ode_secirvvs/parameters_io.h:1130]: its now consistent. Although, we definitly have to reduce the doubling of code in the extrapolation and initialization. Therefore, i have openend issue #794.

**Pull request merged**